internal/cache/file/downloader/job.go

```diff
@@ -463,3 +469,41 @@ func (job *Job) GetStatus() JobStatus {
 	defer job.mu.Unlock()
 	return job.status
 }
+
+func (job *Job) validateChecksum() (err error) {
+	if !job.enableCrcCheck {
+		return
+	}
+
+	crc32Val, err := cacheutil.CalculateFileCRC32(job.fileSpec.Path)
+	if err != nil {
+		return
+	}
+
+	// If checksum matches, simply return.
+	if *job.object.CRC32C == crc32Val {
+		return nil
+	}
+
+	// If the checksum doesn't match there is an error in downloading the object contents.
```

Comment:
Couple of points here:
1. We have made responsibility of CacheHandler to insert or evict into the file cache: https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/cache/file/cache_handler.go#L80. So, we can do the same here i.e. we just fail (plus update the offset in fileInfoCache to 0) the job and let the CacheHandler eventually evict this from cache from next read of same file or when the cache is full. However, I checked and it should be fine to evict (not insert) from job itself.
2. Before evicting the from file cache and after CRC check fail, we should change the state of job to FAILED to prevent others from reusing this file in cache.

Reply:
2. The caller is changing the job status to failed. anyways we are serving the file before the CRC check is completed. so whats wrong in serving the file till the entry is removed?